theia-extensions/viewer-prototype/src/common/trace-viewer-environment.ts

```diff
@@ -3,18 +3,28 @@ import { EnvVariablesServer } from '@theia/core/lib/common/env-variables';
 import { TRACE_SERVER_DEFAULT_URL, TRACE_SERVER_DEFAULT_PORT } from './trace-server-url-provider';
 import { PreferenceService } from '@theia/core/lib/browser';
 import { TRACE_PORT } from '../browser/trace-server-preference';
+import { TraceServerConfigService } from './trace-server-config';
+import { MessageService } from '@theia/core';
 
 @injectable()
 export class TraceViewerEnvironment {
     private port: string | undefined;
 
     constructor(
         @inject(EnvVariablesServer) protected readonly environments: EnvVariablesServer,
-        @inject(PreferenceService) protected readonly preferenceService: PreferenceService) {
+        @inject(PreferenceService) protected readonly preferenceService: PreferenceService,
+        @inject(TraceServerConfigService) protected readonly traceServerConfigService: TraceServerConfigService,
+        @inject(MessageService) protected readonly messageService: MessageService) {
 
         this.port = this.preferenceService.get(TRACE_PORT);
-        this.preferenceService.onPreferenceChanged(event => {
+        this.preferenceService.onPreferenceChanged(async event => {
             if (event.preferenceName === TRACE_PORT) {
+                try {
+                    await this.traceServerConfigService.stopTraceServer();
+                    this.messageService.info(`Trace server disconnected on port: ${this.port}.`);
+                } catch (e){
+                    // Do nothing
```

Comment:
maybe log this?

Reply:
In case if the user modifies the port before starting the server, then we don't want to show any errors :)

Reply:
So comment that :)

Reply:
Done :)